### ISSUE TITLE
Fix injecting one Sass variable inside another variable / path

### ISSUE THREAD

**Issue by MakePixelsWork** [MakePixelsWork/CoreBootstrap#2]
In LESS it is possible to add variables within variables and inject a variable within the path of a folder structure. I've yet to find a solution for this issue for Sass.

Its not a biggy... users are able to update a few instances of code themselves, but its would be handy of this was possible in Sass. This would improve coding a lot. You only need one place to update your framework versioning, and not -like in Font Awesome's case- five updates lines of code.

Example LESS vs Sass vars
![less-vs-sass](https://user-images.githubusercontent.com/56885246/67628440-98c5dc80-f865-11e9-85d6-f94661fbdfd9.png)

**Comment by MakePixelsWork** [comment]
Its a SASS bug, not a bug in my own project. I'm still interested in the solution, which has been part of LESS for years, but does not seem to work in SASS.

**Comment by MakePixelsWork** [comment]
Its actually not really a bug, but a limitation of Sass. Would be nice to have, but not super crucial... especially with the ease of this framework settings. To be continued.

**Comment by MakePixelsWork** [comment]
Interpolation seems to work, but not for every compiler.

https://stackoverflow.com/questions/8608498/have-a-variable-in-images-path-in-sass

**Comment by MakePixelsWork** [comment]
Asked ChatGPT, who said the following:

You're absolutely right that LESS allows variable *interpolation* directly inside strings using the `@{variable}` syntax, which works seamlessly in things like `@import` paths. Sass, on the other hand, **doesn't support this kind of interpolation in `@import` or `@use` paths**, due to how Sass is designed to resolve imports statically (at compile time) for performance and module resolution reasons.

### TL;DR

**No**, you can't dynamically inject a variable into an `@import` or `@use` path in Sass.
But, **there are workarounds**, depending on what you're trying to accomplish.

---

### Why It Doesn’t Work in Sass

In Sass (both SCSS and indented syntax), the `@use` and `@import` directives **must be resolvable statically** (at build time). Variables aren't available during that phase, which means something like this won’t work:

```scss
$version: "v5.3.1";
@use "bootstrap/#{$version}/bootstrap"; // ❌ Doesn't work
```

This will throw an error because Sass cannot interpolate variables into module paths.

---

### Workarounds

#### 1. **Use a Build Tool (like Webpack or Vite)**

If you're using a build tool (like Webpack with `sass-loader`, or Vite), you can conditionally include the correct version by manipulating the include paths or resolving aliases in your config:

```js
// vite.config.js or webpack.config.js
resolve: {
  alias: {
    'bootstrap': path.resolve(__dirname, `node_modules/bootstrap/v5.3.1`)
  }
}
```

Then in your Sass:

```scss
@use "bootstrap/bootstrap";
```

This moves the versioning responsibility to your build system, which is more aligned with Sass's static resolution requirements.

---

#### 2. **Create Separate Entry Files Per Version**

If you're supporting multiple versions, create multiple entry points and choose the one you need programmatically.

```
styles/
├── v5.2.0/
│   └── main.scss
├── v5.3.1/
│   └── main.scss
```

Then import one of these from your main file (or JS logic) based on environment.

---

#### 3. **Use Node-Sass or Dart-Sass with Preprocessing**

You could use a Node.js-based preprocessing step to generate the Sass with the right paths before compiling. For instance:

```js
const fs = require('fs');

const version = 'v5.3.1';
const sassTemplate = `
@use "bootstrap/${version}/bootstrap";
`;

fs.writeFileSync('generated.scss', sassTemplate);
```

Then you compile `generated.scss` with Sass.

---

### Summary

* ✅ **LESS allows path interpolation using variables.**
* ❌ **Sass does not, due to static module resolution.**
* ✅ **You can work around it using build tools, multiple entry files, or pre-build scripts.**

Let me know your stack (e.g., Webpack, Vite, Gulp, etc.), and I can give you a precise solution based on that.
